force-app/main/default/classes/AccountTriggerHandler.cls

```diff
@@ -0,0 +1,84 @@
+public class AccountTriggerHandler extends TriggerHandler {
+
+    private List<Account> newAcctList;
+    private List<Account> oldAcctList;
+    private Map<Id,Account> newAcctMap;
+    private Map<Id,Account> oldAcctMap;
+
+    public AccountTriggerHandler() {
+        
+        this.newAcctList = (List<Account>)Trigger.new;
+        this.oldAcctList = (List<Account>)Trigger.old;
+        this.newAcctMap = (Map<Id,Account>)Trigger.newMap;
+        this.oldAcctMap = (Map<Id,Account>)Trigger.oldMap;
+        //this.setMaxLoopCount(1);
+
+    }
+    public override void beforeInsert() {
+
+        for(Account acc : this.newAcctList) {
+
+            // Set the account type to prospect
+            if (acc.Type == null) {
+                acc.Type = 'Prospect';
+            }
+        
+            // Copy the shipping address to the billing address.
+            if (acc.ShippingStreet != null) {
+                acc.BillingStreet = acc.ShippingStreet;
+            }
+            if (acc.ShippingCity != null) {
+                acc.BillingCity = acc.ShippingCity;
+            }
+            if (acc.ShippingState != null) {
+                acc.BillingState = acc.ShippingState;
+            }
+            if (acc.ShippingPostalCode != null) {
+                acc.BillingPostalCode = acc.ShippingPostalCode;
+            }
+            if (acc.ShippingCountry != null) {
+                acc.BillingCountry = acc.ShippingCountry;
+            }
+
+            // Set the account rating to 'Hot' if the Phone, Website, and Fax is not empty.
+            if (acc.Phone != null && acc.Website != null && acc.Fax != null) {
+                acc.Rating = 'Hot';
+            }
+        }
```

Comment:
Adding your logic directly to the onInsert method will get very hard to maintain over time. 

My recommendation is that you call the methods you need from here.  Something like what is below. 

```
  public override void beforeInsert(){
        setDefaultAccountType();
        setDefaultBillingAddress();
        setRating();
    }
    ```